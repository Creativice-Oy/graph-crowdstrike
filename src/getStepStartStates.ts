import {
  IntegrationExecutionContext,
  StepStartStates,
} from '@jupiterone/integration-sdk-core';
import { CrowdStrikeIntegrationInstanceConfig } from './types';
import { fetchVulnerabilitiesStep } from './steps/fetchVulnerabilities';
import { fetchDevicesStep } from './steps/fetchDevices';
import { fetchDevicePolicyRelationshipsStep } from './steps/fetchDevicePolicyRelationships';
import { fetchPreventionPoliciesStep } from './steps/fetchPreventionPolicies';
import { getAccountStep } from './steps/getAccount';

/**
 * Introduced to allow for specific accounts to trial the vulnerabilities step.
 * TODO: Remove me once vuln step has been proven.
 * @param executionContext
 */
export default function getStepStartStates(
  executionContext: IntegrationExecutionContext<CrowdStrikeIntegrationInstanceConfig>,
): StepStartStates {
  return {
    [getAccountStep.id]: { disabled: false },
    [fetchDevicesStep.id]: { disabled: false },
    [fetchDevicePolicyRelationshipsStep.id]: { disabled: false },
    [fetchPreventionPoliciesStep.id]: { disabled: false },
    [fetchVulnerabilitiesStep.id]: {
      disabled:
        !(executionContext.instance.config as any)
          .enableFetchVulnerabilitiesStep ?? true,
    },
  };
}
