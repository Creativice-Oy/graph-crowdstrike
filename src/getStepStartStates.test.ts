import getStepStartStates from './getStepStartStates';

describe('getStepStartStates', () => {
  test('it returns correct disabled values', () => {
    const executionContext = {
      instance: {
        config: {
          enableFetchVulnerabilitiesStep: true,
        },
      },
    };

    expect(getStepStartStates(executionContext as any)).toMatchSnapshot();

    const executionContext2 = {
      instance: {
        config: {},
      },
    };
    expect(getStepStartStates(executionContext2 as any)).toMatchSnapshot();
  });
});
