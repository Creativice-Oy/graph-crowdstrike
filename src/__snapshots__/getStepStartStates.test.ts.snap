// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStepStartStates it returns correct disabled values 1`] = `
Object {
  "fetch-device-policies": Object {
    "disabled": false,
  },
  "fetch-devices": Object {
    "disabled": false,
  },
  "fetch-prevention-policies": Object {
    "disabled": false,
  },
  "fetch-vulnerabilities": Object {
    "disabled": false,
  },
  "get-account": Object {
    "disabled": false,
  },
}
`;

exports[`getStepStartStates it returns correct disabled values 2`] = `
Object {
  "fetch-device-policies": Object {
    "disabled": false,
  },
  "fetch-devices": Object {
    "disabled": false,
  },
  "fetch-prevention-policies": Object {
    "disabled": false,
  },
  "fetch-vulnerabilities": Object {
    "disabled": true,
  },
  "get-account": Object {
    "disabled": false,
  },
}
`;
